### PR TITLE
Unit Testing on Login Component

### PR DESCRIPTION
**Pull request by Monfernape**
## Change Request
Write a unit testing for a client side component 

### Implementation

- Added `vitest` and all related testing libraries
- Created a `vitest.config.ts` file for the configuration of the `vitest`

### Resources
Test Pass
image
Test Fail
image

### CODE REVIEW

**Comment by Monfernape** [comment]
Mubashir, ticket's description is wrong. It should mention that this PR adds the configuration to support unit testing in the app.

File: src/app/__test__/Login.test.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { expect, test } from "vitest";
```
Comment (Monfernape) [inline src/app/__test__/Login.test.tsx:1]: I'm not sure if it makes sense to have a separate directory for tests. Wdyt? IMO should be kept close to component origin.

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
-import { redirect } from 'next/navigation'
+import { redirect } from "next/navigation";
 
 export default function Home() {
   const isUserLogedIn = false;
   if (!isUserLogedIn) {
-    redirect('/login')
+    redirect("/login");
```
Comment (Monfernape) [inline src/app/page.tsx:6]: Once Nadir's PR for layout is merged, this should use enum.

**Review by Monfernape — COMMENTED** [review]
Looks great Mubashir.

File: src/app/__test__/Login.test.tsx (hunk 1 above)
Comment (Monfernape) [reply src/app/__test__/Login.test.tsx:1]: it is not necessary to have a separate directory for testing.so i kept it in __test__ folder but now i decided to keep it with component origin

**Review by ghulamhafeez — APPROVED** [review]

**Pull request merged**